lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -845,6 +860,30 @@ export abstract class ClientApplication {
         }
     }
 
+    /**
+     * Returns the first account found in the cache that matches the account filter passed in.
+     * @param accountFilter 
+     * @returns The first account found in the cache matching the provided filter or null if no account could be found.
+     */
+    getAccountByFilter(accountFilter: AccountFilter): AccountInfo | null {
```

Comment:
Do we need both getAccount and getAccounts or can we get by with just one?

Reply:
Here's how I see it:

- getAccountByFilter - Replaces current getAccountBy APIs so they can be deprecated
- getAccountsByFilter - Allows devs to leverage MSAL's existing account filtering and (I think more importantly) enables  powerful actions in a multi-tenant scenario like getting truncated home account objects that contain their cross-tenant profiles (instead of getting an exhaustive array of every account info object in the cache each with every cross-tenant profile represented)

Reply:
Removed `getAccountsByFilter` since it was incorporated into `getAllAccounts` with an optional filter